services/OneShipStation_XmlService.php

```diff
@@ -340,7 +340,12 @@ public function customOrderFields(\SimpleXMLElement $order_xml, Commerce_OrderMo
                 foreach ($customFieldCallbacks as $callback) {
                     if (is_callable($callback)) {
                         $value = $callback($order);
-                        $order_xml->addChild($fieldName, substr(htmlspecialchars($value), 0, 100));
+                        if (strpos($fieldName, 'CustomField') !== false) {
+                            $fieldValue = substr(htmlspecialchars($value), 0, 100);
+                        } else {
+                            $fieldValue = htmlspecialchars($value);
```

Comment:
To be safe, we should probably restrict this to 1000 chars.

Reply:
:+1: